packages/core/src/components/overlay/overlay.tsx

```diff
@@ -475,25 +553,14 @@ export class Overlay extends AbstractPureComponent2<OverlayProps, IOverlayState>
                 return elem && elem.contains(eventTarget) && !elem.isSameNode(eventTarget);
             });
 
-        if (isOpen && canOutsideClickClose && !isClickInThisOverlayOrDescendant) {
-            // casting to any because this is a native event
-            onClose?.(e as any);
-        }
-    };
-
-    private handleDocumentFocus = (e: FocusEvent) => {
-        // get the actually target even if we are in an open mode Shadow DOM
-        const eventTarget = e.composed ? e.composedPath()[0] : e.target;
-        if (
-            this.props.enforceFocus &&
-            this.containerElement != null &&
-            eventTarget instanceof Node &&
-            !this.containerElement.contains(eventTarget as HTMLElement)
-        ) {
-            // prevent default focus behavior (sometimes auto-scrolls the page)
-            e.preventDefault();
-            e.stopImmediatePropagation();
-            this.bringFocusInsideOverlay();
```

Comment:
when `enforceFocus=true`, focus will start on the overlay and will only leave if another overlay with autoFocus or enforceFocus is opened. because we take care to focus the last opened overlay when closing, we don't need this handler anymore